src/openforms/js/components/formio_builder/builder.js

```diff
@@ -368,15 +368,18 @@ const FormIOBuilder = ({
     const exposeTranslations = translations => {
       // update the component form submission data, so we have the updated translations
       // in the component.
-      instance.root.submission = {
-        data: {
-          ...newComponentConfiguration,
-          openForms: {
-            ...(newComponentConfiguration?.openForms || {}),
-            translations: translations,
+      instance.root.setSubmission(
+        {
+          data: {
+            ...newComponentConfiguration,
+            openForms: {
+              ...(newComponentConfiguration?.openForms || {}),
+              translations: translations,
+            },
           },
         },
-      };
+        {fromSubmission: false}
```

Comment:
It seems that here https://github.com/formio/formio.js/blob/v4.13.13/src/components/_classes/component/Component.js#L2632
formio does this check on `fromSubmission`. Before we added the custom `onChange` for the translations, the `fromSubmission` flag was undefined. The place where this flag gets set is the `setSubmission` function https://github.com/formio/formio.js/blob/v4.13.13/src/Webform.js#L783. 
Not really sure yet what this `fromSubmission` function means, but this seems to solve our problem

Reply:
I think it tracks if there was existing data that was loaded (like when you resume a paused form in our SDK), so then it won't override the submission data with calculated data because it was persisted already at some point.

I can't find any references to it in the documentation, only in the source code, but it seems that setting it to false here would be correct usage because we're doing programmatic stuff behind the scenes.

Reply:
The problem with setting the flag here to false is that then I think the component is not updated, so the translations are no longer updated.

The `setValue` works better than the `setSubmission`, but any new translations are not added. I think it may be a matter of findingin which order the different updates have to happen so that the component is rendered with all the updates :see_no_evil: